### DISCUSSION TITLE
How to deal with database is locked errors?

### DISCUSSION THREAD

**Discussion by zareami10** [dagster-io/dagster#8617] [Q&A] [answered]
I keep getting many errors mentioning the database is locked on various occasions, for example the latest one being:

```
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) database is locked
[SQL: UPDATE runs SET status=?, run_body=?, update_timestamp=?, end_time=? WHERE runs.run_id = ?]
[parameters: ('FAILURE', '{"__class__": "PipelineRun", "asset_selection": {"__frozenset__": [{"__class__": "AssetKey", "path": ["**"]}]}, "execution_plan_snaps ... (1759 characters truncated) ... <parts removed>
(Background on this error at: https://sqlalche.me/e/14/e3q8)
```

How should I recover from such errors? I'm not even sure where the mentioned db is located at.

**Comment by zareami10** [comment, marked as answer]
Switching to PostgreSQL greatly improved the situation.

**Reply by sryza** [reply]
That makes sense. For the record, the issue is that sqlite doesn't support concurrent access.

**Reply by zareami10** [reply]
> That makes sense. For the record, the issue is that sqlite doesn't support concurrent access.

That also makes the whole experience with dagit for example quite slow and unreliable, giving a bit of a bad first impression. Maybe the DB situation is worth mentioning in the installation docs?

**Reply by rubenbriones** [reply]
I'm having the same problem with SQLite. ¿It is possible to migrate SQLite storage to postgres storage? 
Thanks.

**Reply by dmsfabiano** [reply]
Even though SQLite does not support concurrent access, sqlalchemy does support [thread-safe sessions](https://docs.sqlalchemy.org/en/20/orm/contextual.html). Still I am not seeing away to share the sessions that get passed to concurrent ops, because the returns of an op have to be pickable, any ideas?
